src/server/pps/server/worker_rc.go

```diff
@@ -77,6 +79,20 @@ func (a *apiServer) workerPodSpec(options *workerOptions) api.PodSpec {
 		sidecarVolumeMounts = append(sidecarVolumeMounts, secretMount)
 		userVolumeMounts = append(userVolumeMounts, secretMount)
 	}
+
+	options.volumes = append(options.volumes, api.Volume{
```

Comment:
Why do we need `docker.sock`?

Reply:
To detect the `user`, `working_dir` and `entry_point`, that's all done via a call to `docker inspect`.